verilog/parser/verilog.lex

```diff
@@ -672,6 +672,7 @@ wait_order { UpdateLocation(); return TK_wait_order; }
 wildcard { UpdateLocation(); return TK_wildcard; }
 <COVERGROUP>with { UpdateLocation(); return TK_with__covergroup; }
 with { UpdateLocation(); return TK_with; }
+"with"[ \t]*"[" { UpdateLocation(); return TK_with_followed_by_bracket; }
```

Comment:
I would avoid merging this as a single token, however you can achieve context-sensitivity to give variants of the same textual token by using a combination of lexer states and `yyless` to rewind.  Keep in mind that when you create variants of tokens like `[` and `(` (and yes, even `;`) that the formatter needs to be handle these similarly.

Reply:
Thanks, I think I'll reformulate it like that.

Reply:
Another thing to keep in mind is *comments*.  You'll notice that in various state blocks in the lexer, I had to explicitly add comment handling to push state.